server/routes/polls/polls.route.js

```diff
@@ -2,82 +2,97 @@ const express = require('express');
 const router = express.Router();
 const Poll = require('../../models/polls');
 const upload = require('./utils');
+const passport = require('passport');
 
 // @route POST /polls
 // @desc To add a new poll
 // @access private
 // @params title, userId, imagess x 2
-router.post('/', upload.any(), (req, res) => {
-  const errors = [];
-  if (!req.body.title) errors.push({ title: 'Invalid title' });
-  if (!req.body.userId) errors.push({ userId: 'Invalid user id' });
-  if (req.files.length != 2)
-    errors.push({ images: 'minimum two images required' });
+router.post(
+  '/',
+  passport.authenticate('jwt', { session: false }),
+  upload.any(),
+  (req, res) => {
+    const errors = [];
+    if (!req.body.title) errors.push({ title: 'Invalid title' });
+    if (req.files.length != 2)
+      errors.push({ images: 'minimum two images required' });
 
-  if (errors.length > 0) return res.status(400).json({ error: errors });
-  const poll = new Poll({
-    title: req.body.title,
-    userId: req.body.userId,
-  });
-  // Trying to save the poll
-  try {
-    poll.save();
-  } catch (error) {
-    return res.status(400).json({ error: error.message });
-  }
-  // Saving file location to the poll object
-  try {
-    req.files.map((item) => {
-      poll.addImage(item.location);
+    if (errors.length > 0) return res.status(400).json({ error: errors });
+    const poll = new Poll({
+      title: req.body.title,
+      userId: req.user.id,
     });
-  } catch (error) {
-    return res.status(400).json({ error: error.message });
+    // Trying to save the poll
+    try {
+      poll.save();
+    } catch (error) {
+      return res.status(400).json({ error: error.message });
+    }
+    // Saving file location to the poll object
+    try {
+      req.files.map((item) => {
+        poll.addImage(item.location);
+      });
+    } catch (error) {
+      return res.status(400).json({ error: error.message });
+    }
+    return res.status(200).json({ poll: poll });
   }
-  return res.status(200).json({ poll: poll });
-});
+);
 
 // @route DELETE "/polls"
 // @desc to be delete a poll with a given pollId
 // @params pollId
 // @access private
-router.delete('/:pollId', (req, res) => {
-  const pollId = req.params['pollId'];
-  Poll.findById(pollId, function (err, poll) {
-    if (err) return res.status(400).json({ error: 'poll not found' });
-    else poll.remove();
-    return res.status(200).json({ status: 'ok' });
-  });
-});
+router.delete(
+  '/:pollId',
+  passport.authenticate('jwt', { session: false }),
+  (req, res) => {
+    const pollId = req.params['pollId'];
+    Poll.findById(pollId, function (err, poll) {
```

Comment:
For these routes below, you need to confirm that the req.user is the owner of these polls